src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedIntrinsicOperatorSymbol.cs

```diff
@@ -22,6 +22,7 @@ internal sealed class SynthesizedIntrinsicOperatorSymbol : MethodSymbol
 
         public SynthesizedIntrinsicOperatorSymbol(TypeSymbol leftType, string name, TypeSymbol rightType, TypeSymbol returnType, bool isCheckedBuiltin)
         {
+            Debug.Assert(SyntaxFacts.IsCheckedOperator(name) == isCheckedBuiltin);
```

Comment:
>SyntaxFacts.IsCheckedOperator(name) == isCheckedBuiltin [](http://example.com/codeflow?start=25&length=55)

This assumption might not hold true for operators that do not have checked form. For cases like this, we should simply adjust the boolean to reflect the fact. Doing that here should be fine.

Reply:
hrmm... that surprises me.  The way we get the name at the callsite is to feed this same boolean into the helper that produces the name.  So i would expect these to match.

Reply:
basically, teh caller does this:

```c#
                    symbols = ImmutableArray.Create<Symbol>(new SynthesizedIntrinsicOperatorSymbol(objectType,
                                                                                             OperatorFacts.BinaryOperatorNameFromOperatorKind(op, isChecked: binaryOperator.OperatorKind.IsChecked()),
                                                                                             objectType,
                                                                                             binaryOperator.Type,
                                                                                             binaryOperator.OperatorKind.IsChecked()));
```

(or the unary equivalent).  So the same `binaryOperator.OperatorKind.IsChecked()` feeds into determining the name *and* passing the isCheckedBuiltIn flag through.  

We could even consider not passing the boolean in and just determining it from the name if that works for you.

Reply:
> We could even consider not passing the boolean in and just determining it from the name if that works for you.

Yes, that would work

Reply:
Done.

Note: tehre was an interesting change here (but as this only affects tehse 'ide' symbols i think it's fine).  Specifically, it used to be possible to get 'IsCheckedBuiltin=true' methods even for a non-checked method name.  This would happen with virtually all dynamic operators, even for operators that have no checked version.  

Now, by using hte name, the checked-ness or not is based only on the name alone.  This led to some changes in test behavior, but i think it's acceptable/appropriate.